src/app/EventManagement.cpp

```diff
@@ -522,38 +522,45 @@ CHIP_ERROR EventManagement::CopyEvent(const TLVReader & aReader, TLVWriter & aWr
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR EventManagement::CheckEventContext(EventLoadOutContext * eventLoadOutContext,
-                                              const EventManagement::EventEnvelopeContext & event)
+bool EventManagement::CheckEventContext(EventLoadOutContext * eventLoadOutContext,
+                                        const EventManagement::EventEnvelopeContext & event)
 {
     if (eventLoadOutContext->mCurrentEventNumber < eventLoadOutContext->mStartingEventNumber)
     {
-        return CHIP_ERROR_UNEXPECTED_EVENT;
+        return false;
     }
 
     if (event.mFabricIndex.HasValue() &&
         (event.mFabricIndex.Value() == kUndefinedFabricIndex ||
          eventLoadOutContext->mSubjectDescriptor.fabricIndex != event.mFabricIndex.Value()))
     {
-        return CHIP_ERROR_UNEXPECTED_EVENT;
+        return false;
     }
 
+    bool isPathInterested = false;
     ConcreteEventPath path(event.mEndpointId, event.mClusterId, event.mEventId);
-    CHIP_ERROR ret = CHIP_ERROR_UNEXPECTED_EVENT;
-
+    // Check whether the event path is in the interested paths
     for (auto * interestedPath = eventLoadOutContext->mpInterestedEventPaths; interestedPath != nullptr;
          interestedPath        = interestedPath->mpNext)
     {
         if (interestedPath->mValue.IsEventPathSupersetOf(path))
         {
-            ret = CHIP_NO_ERROR;
+            isPathInterested = true;
             break;
         }
     }
-
-    ReturnErrorOnFailure(ret);
+    if (!isPathInterested)
+    {
+        return false;
+    }
 
     DataModel::EventEntry eventInfo;
-    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->GetDataModelProvider()->EventInfo(path, eventInfo));
+    if (InteractionModelEngine::GetInstance()->GetDataModelProvider()->EventInfo(path, eventInfo) != CHIP_NO_ERROR)
+    {
+        // If we cannot get event info here, the interested path should be a wildcard path but the data model does't
+        // support the event, return false here so the event path will be excluded in the generated event report.
```

Comment:
So just to make sure: this is an event that:

1.  Our data model claims we don't support.
2. We actually generated an instance of.

I have to ask: why do we even have this event in our buffer?   It's taking up space, possibly evicting other events, but we will never actually deliver it to anyone, right?

As in, should an existence/support check perhaps be where events are placed in the buffer?

That does still leave the question of what to do if that check passed but this one fails.  That could in fact happen if the endpoint the event was for has gone away, right?  Though ideally in that situation we would remove those stray events for the now-gone endpoint from the buffer (another followup may be needed here).

But for now we're basically preserving behavior here, ok.

Reply:
This issue is not for the code-gen data model provider, since our code-gen data model provider support all the events. It occurs on a data model provider that doesn't support some optional events. If a client reads a wildcard event path that includes the unsupported events, the read will fail, instead of not including the unsupported events in the report.